index.bs

```diff
@@ -3808,18 +4263,388 @@ partial dictionary MLOpSupportLimits {
+Gather values of the input tensor along an axis according to the indices.
+
+<script type=idl>
+partial interface MLGraphBuilder {
+  MLOperand gatherElements(MLOperand input,
+                           MLOperand indices,
+                           optional MLGatherOptions options = {});
+};
+
+partial dictionary MLOpSupportLimits {
+  MLGatherSupportLimits gatherElements;
+};
+</script>
+
+<div dfn-for="MLGraphBuilder/gatherElements(input, indices, options)" dfn-type=argument>
+    **Arguments:**
+        - <dfn>input</dfn>: an {{MLOperand}}. The input N-D tensor from which the values are gathered.
+        - <dfn>indices</dfn>: an {{MLOperand}}. The indices N-D tensor of the input values to gather. The values must be of type {{MLOperandDataType/"int32"}}, {{MLOperandDataType/"uint32"}}, or {{MLOperandDataType/"int64"}}, and must be in the range -N (inclusive) to N (exclusive) where N is the size of the input dimension indexed by *options.axis*, and a negative index means indexing from the end of the dimension.
+        - <dfn>options</dfn>: an optional {{MLGatherOptions}}. The optional parameters of the operation.
+
+    **Returns:** an {{MLOperand}}. The output N-D tensor of [=MLOperand/rank=] equal to the [=MLOperand/rank=] of *input*.
+</div>
+
+<table id=constraints-gatherelements class='data' link-for="MLGraphBuilder/gatherElements(input, indices, options)">
+  <caption>Constraints for {{MLGraphBuilder/gatherElements()}}</caption>
+  <thead>
+    <tr>
+      <th>operand</th>
+      <th>[=/allowed data types=]</th>
+      <th>[=/allowed ranks=]</th>
+    </tr>
+  </thead>
+  <tr>
+    <td>{{input}}</td>
+    <td>[=/any data type|any=]</td>
+    <td>1 to [=/any rank|N=]</td>
+  </tr>
+  <tr>
+    <td>{{indices}}</td>
+    <td>{{MLOperandDataType/"int32"}}, {{MLOperandDataType/"uint32"}}, {{MLOperandDataType/"int64"}}</td>
+    <td>*input*'s [=MLOperand/rank=]</td>
+  </tr>
+  <tr>
+    <td>*output*</td>
+    <td>[=/same type as|same as=] {{input}}</td>
+    <td>*input*'s [=MLOperand/rank=]</td>
+  </tr>
+</table>
+
+{{MLOpSupportLimits}} has the following members for {{MLGraphBuilder/gatherElements()}}:
+<dl dfn-type=dict-member dfn-for=MLOpSupportLimits>
+    : <dfn>gatherElements</dfn>
+    :: Support limits for operator {{MLGraphBuilder/gatherElements()}}.
+</dl>
+
+<div class="note">
+  The {{MLGraphBuilder/gatherElements(input, indices, options)/indices}} parameter to {{MLGraphBuilder/gatherElements()}} can not be clamped to the allowed range when the graph is built because the inputs are not known until execution. Implementations can introduce {{MLGraphBuilder/clamp()}} in the compiled graph if the required clamping behavior is not provided by the underlying platform. Similarly, if the underlying platform does not support negative indices, the implementation can introduce operations in the compiled graph to transform a negative index from the end of the dimension into a positive index.
```

Comment:
Address RFC2119 warning:
```suggestion
  The {{MLGraphBuilder/gatherElements(input, indices, options)/indices}} parameter to {{MLGraphBuilder/gatherElements()}} can not be clamped to the allowed range when the graph is built because the inputs are not known until execution. Implementations can introduce {{MLGraphBuilder/clamp()}} in the compiled graph if the specified clamping behavior is not provided by the underlying platform. Similarly, if the underlying platform does not support negative indices, the implementation can introduce operations in the compiled graph to transform a negative index from the end of the dimension into a positive index.
```